applications/CoSimulationApplication/python_scripts/data_transfer_operators/kratos_mapping.py

```diff
@@ -37,27 +48,35 @@ def __init__(self, settings, parent_coupled_solver_data_communicator):
         if not settings.Has("mapper_settings"):
             raise Exception('No "mapper_settings" provided!')
         super().__init__(settings, parent_coupled_solver_data_communicator)
+        self.debug_vtk = self.settings["debug_vtk"].GetBool()
+        self.auxiliary_debug_counter = self.settings["auxiliary_debug_counter"].GetBool()
         self.__mappers = {}
+        self.sub_counter = 0
+        self.number_of_mappers = 0
 
     def _ExecuteTransferData(self, from_solver_data, to_solver_data, transfer_options):
-        model_part_origin_name = from_solver_data.model_part_name
-        variable_origin        = from_solver_data.variable
-        identifier_origin      = from_solver_data.solver_name + "." + model_part_origin_name
-
-        model_part_destination_name = to_solver_data.model_part_name
-        variable_destination        = to_solver_data.variable
-        identifier_destination      = to_solver_data.solver_name + "." + model_part_destination_name
-
-        mapper_flags = self.__GetMapperFlags(transfer_options, from_solver_data, to_solver_data)
-
-        identifier_tuple         = (identifier_origin, identifier_destination)
-        inverse_identifier_tuple = (identifier_destination, identifier_origin)
+        # Prepare the solver data
+        model_part_origin, model_part_destination, model_part_origin_name, model_part_destination_name, variable_origin, variable_destination, mapper_flags, identifier_origin, identifier_destination, identifier_tuple, inverse_identifier_tuple = self.__PrepareSolverData(from_solver_data, to_solver_data, transfer_options)
```

Comment:
11 return arguments :sweat_smile:

Reply:
![flat,750x,075,f-pad,750x1000,f8f8f8](https://github.com/user-attachments/assets/3ed8ab7d-6d3d-4e6c-935d-fb97045caed1)

Reply:
If I move to a new class I can refactor this and make it cleaner perhaps